apps/dashboard/src/contract-ui/tabs/nfts/components/airdrop-tab.tsx

```diff
@@ -43,44 +59,67 @@ const AirdropTab: React.FC<AirdropTabProps> = ({ contract, tokenId }) => {
   return (
     <Stack w="full">
       <form
-        onSubmit={handleSubmit((data) => {
+        onSubmit={handleSubmit(async (_data) => {
           trackEvent({
             category: "nft",
             action: "airdrop",
             label: "attempt",
-            contractAddress: contract?.getAddress(),
+            contractAddress,
             token_id: tokenId,
           });
-          airdrop.mutate(
-            {
-              tokenId,
-              addresses: data.addresses,
+          const totalOwned = await balanceOf({
+            contract,
+            tokenId: BigInt(tokenId),
+            owner: account?.address ?? "",
+          });
+          // todo: make a batch-transfer extension for erc1155?
+          const totalToAirdrop = _data.addresses.reduce((prev, curr) => {
+            return BigInt(prev) + BigInt(curr?.quantity || 1);
+          }, 0n);
+          if (totalOwned < totalToAirdrop) {
+            throw new Error(
+              `The caller owns ${totalOwned.toString()} NFTs, but wants to airdrop ${totalToAirdrop.toString()} NFTs.`,
+            );
```

Comment:
when throwing here will this not crash the page? where is this being caught?

Reply:
Yeah I will probably return with a toast here